pkg/scope/provider.go

```diff
@@ -45,9 +48,11 @@ const (
 	caSecretKey     = "cacert"
 )
 
-type providerScopeFactory struct{}
+type providerScopeFactory struct {
+	cache *cache.LRUExpireCache
```

Comment:
We could make this cache more explicit, by e.g. naming it `clientCache`.
```suggestion
	clientCache *cache.LRUExpireCache
```